src/kaleidoscope/hardware/kbdfans/KBD4x.h

```diff
@@ -45,6 +45,11 @@ class KBD4x: public kaleidoscope::hardware::ATMegaKeyboard {
 
   static constexpr int8_t led_count = 0;
 
+  typedef MatrixAddr<matrix_rows, matrix_columns> KeyAddr;
+  static constexpr int8_t numKeys() {
+    return matrix_columns * matrix_rows;
+  }
+
```

Comment:
Can this get lifted up to ATMegaKeyboard? Part of the point of having the abstraction there is to reduce the boilerplate code for each implementation.

Reply:
Not yet. Currently the base class `ATMegaKeyboard` is unaware of the dimensions of the derived keyboards' key matrices. 

But I think that's something, @algernon is currently working on.

Reply:
Is that the changes in the bootloader PR? If so, I'd love to get that merged in and this updated for that.

Reply:
Kiiiiinda. #611 makes the derived keyboard accessible to the base class, so yeah, it will be - or can be made - aware of the keyboard dimensions.